CMakeLists.txt

```diff
@@ -62,6 +62,10 @@ set(COMPLETE_VERSION ${CPACK_PACKAGE_VERSION_MAJOR}.${CPACK_PACKAGE_VERSION_MINO
 set(RELEASE_NAME "Master")
 project(qgis VERSION ${COMPLETE_VERSION})
 
+if(NOT CMAKE_SIZEOF_VOID_P EQUAL 8)
```

Comment:
More an educational comment than anything, but there are architectures, like CHERI, with 128-bit pointers: https://lwn.net/Articles/909265/ , so if we don't want to exclude them but just 32-bit, the test could be "CMAKE_SIZEOF_VOID_P EQUAL 4"
I've experimented that a bit with GDAL (for fun) ~ 2 years ago, and that mostly worked, except in all places where one serializes a pointer to text ( printf("%p) ) and deserialize it back to pointer ( sscanf("%p)" ).